rust/libceed/src/vector.rs

```diff
@@ -375,6 +398,80 @@ impl<'a> Vector<'a> {
         self.check_error(ierr)
     }
 
+    /// Wrap a mutable slice in a Vector of the same length
+    ///
+    /// # arguments
+    ///
+    /// * `slice` - values to wrap in self; length must match
+    ///
+    /// ```
+    /// # use libceed::prelude::*;
+    /// # fn main() -> libceed::Result<()> {
+    /// # let ceed = libceed::Ceed::default_init();
+    /// let mut vec = ceed.vector(4)?;
+    /// let mut array = [10., 11., 12., 13.];
+    ///
+    /// {
+    ///     // `wrapper` holds a mutable reference to the wrapped slice
+    ///     //   that is dropped when `wrapper` goes out of scope
+    ///     let wrapper = vec.wrap_slice_mut(&mut array)?;
+    ///     vec.view()?.iter().enumerate().for_each(|(i, v)| {
+    ///         assert_eq!(*v, 10. + i as Scalar, "Slice not set correctly");
+    ///     });
+    ///
+    ///     // This line will not compile, as the `wrapper` holds mutable
+    ///     //   access to the `array`
+    ///     // array[0] = 5.0;
+    ///
+    ///     // Changes here are reflected in the `array`
+    ///     vec.set_value(5.0)?;
+    ///     vec.view()?.iter().for_each(|v| {
+    ///         assert_eq!(*v, 5.0 as Scalar, "Value not set correctly");
+    ///     });
+    /// }
+    ///
+    /// // 'array' remains changed
+    /// array.iter().for_each(|v| {
+    ///     assert_eq!(*v, 5.0 as Scalar, "Array not mutated correctly");
+    /// });
+    ///
+    /// // While changes to `vec` no longer affect `array`
+    /// vec.set_value(6.0)?;
+    /// array.iter().for_each(|v| {
+    ///     assert_eq!(*v, 5.0 as Scalar, "Array mutated without permission");
+    /// });
```

Comment:
Do you prefer the above style? You seem to be using it consistently, though this alternative has slightly less syntactic noise.
```suggestion
    /// for v in array.iter() {
    ///     assert_eq!(*v, 5.0 as Scalar, "Array mutated without permission");
    /// }
```
From [the docs](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each)
> This is equivalent to using a for loop on the iterator, although break and continue are not possible from a closure. It’s generally more idiomatic to use a for loop, but for_each may be more legible when processing items at the end of longer iterator chains. In some cases for_each may also be faster than a loop, because it will use internal iteration on adapters like Chain.

Reply:
No particular reason, I'm just not used to mixing and matching an iterator and older style loops

Reply:
I switched to `for ... in` when the `.for_each` isn't at the 'end of longer iterator chains'.